addons/l10n_be_hr_payroll/models/l10n_be_hr_payroll.py

```diff
@@ -1,9 +1,90 @@
+from datetime import date, datetime
+from pytz import timezone
 from odoo import api, fields, models, _
-
 from odoo.exceptions import ValidationError
+from odoo.osv.expression import AND
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+
+    def _get_13th_month_basic(self, struct_cp_200):
+        contracts = self.employee_id.contract_ids.filtered(lambda c: c.state not in ['draft', 'cancel'] and c.struct_id == struct_cp_200)
+        if not contracts:
+            return 0.0
+
+        # 1. Seniority: an employee must have a seniority of at least 6 months to be granted the 13th month
+        first_contract = min(contracts, key=lambda c: c.date_start)
+        start_of_service = first_contract.date_start
+        year = self.date_to.year
+        if start_of_service >= date(year, 7, 1):
+            return 0.0
+
+        # 2. Working months
+        n_months = 12
+        if start_of_service > date(year, 1, 1):
+            n_months = 12 - start_of_service.month
+            # only count full months
+            work_days = {int(d) for d in first_contract.resource_calendar_id.mapped('normal_attendance_ids.dayofweek')}
+            # working days between the 1st and the start of service (including start of service)
+            first_days_of_month = rrule.rrule(rrule.DAILY, byweekday=work_days, dtstart=start_of_service.replace(day=1), until=start_of_service)
+            if len(list(first_days_of_month)) <= 1:
+                n_months += 1
+
+        # 3. Deduct absences
+        unpaid_benefit_type = self.env['hr.benefit.type']._load_from_xmlid(
+            'hr_payroll.benefit_type_unpaid_leave',
+            'l10n_be_hr_payroll.benefit_type_credit',
+            'l10n_be_hr_payroll.benefit_type_unpredictable',
+            'l10n_be_hr_payroll.benefit_type_long_sick',
+            'l10n_be_hr_payroll.benefit_type_part_sick',
+        )
+        paid_benefit_type = self.env['hr.benefit.type']._load_from_xmlid(
+            'hr_payroll.benefit_type_attendance',
+            'hr_payroll.benefit_type_home_working',
+            'hr_payroll.benefit_type_legal_leave',
+            'hr_payroll.benefit_type_sick_leave',
+            'l10n_be_hr_payroll.benefit_type_additional_paid',
+            'l10n_be_hr_payroll.benefit_type_breast_feeding',
+            'l10n_be_hr_payroll.benefit_type_european',
+            'l10n_be_hr_payroll.benefit_type_extra_legal',
+            'l10n_be_hr_payroll.benefit_type_maternity',
+            'l10n_be_hr_payroll.benefit_type_notice',
+            'l10n_be_hr_payroll.benefit_type_paternity_company',
+            'l10n_be_hr_payroll.benefit_type_paternity_legal',
+            'l10n_be_hr_payroll.benefit_type_recovery',
+            'l10n_be_hr_payroll.benefit_type_recovery_additional',
+            'l10n_be_hr_payroll.benefit_type_small_unemployment',
+            'l10n_be_hr_payroll.benefit_type_training',
+            'l10n_be_hr_payroll.benefit_type_training_time_off',
+        )
```

Comment:
What about adding two m2m on salary structures: `paid_work_entry_type` and `unpaid_work_entry_type` ?
There are different for many structures (13th month, double holiday, "indemnités de rupture", etc)
It would avoid hard coding them like here each time.
And it is more flexible than adding an `unpaid` boolean field like in this PR (commit cfd7981)